### DISCUSSION TITLE
Multiple API requests in getserversideprops will degrade performance - Is there an optimization method

### DISCUSSION THREAD

**Discussion by parlay96** [vercel/next.js#34672] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

no

### What browser are you using? (if relevant)

chrome

### How are you deploying your application? (if relevant)

next start

### Describe the Bug

Multiple API requests in getserversideprops will degrade performance - next js

  1.  In addition: if there are many data interfaces on the server side. A lot of data is returned. This will cause the page rendering to be particularly slow !
   2.   I want to know if there is any way to optimize it, because the demand of our project is that the whole page needs SEO and needs to be rendered in advance. Cannot client render!

### Expected Behavior

Slow rendering

### To Reproduce

Multiple API requests in getserversideprops will degrade performance - next js

  1.  In addition: if there are many data interfaces on the server side. A lot of data is returned. This will cause the page rendering to be particularly slow !
   2.   I want to know if there is any way to optimize it, because the demand of our project is that the whole page needs SEO and needs to be rendered in advance. Cannot client render!
 3. Requirements design all requests are server-side, not client-side, need SEO

**Comment by parlay96** [comment]
This is not a next bug issue， But I really want to solve it

**Comment by parlay96** [comment]
Does anyone know?

**Comment by balazsorban44** [comment]
If you are optimizing for SEO, I assume the content you are loading can be public. For that case, `getStaticProps` might be a better alternative as the output can be cached for subsequent requests, while also calling slower APIs less: https://nextjs.org/docs/basic-features/data-fetching/get-static-props

**Comment by parlay96** [comment]
@balazsorban44  I'm also a novice,  Can I know the difference between getserversideprops and getstaticprops. Because we need server-side rendering, we choose getserversideprops. We need to refresh the page to get new data,

**Reply by balazsorban44** [reply]
Please refer to the documentation.

https://nextjs.org/docs/basic-features/data-fetching/overview

https://nextjs.org/learn/basics/data-fetching

**Reply by parlay96** [reply]
On the other hand, [Static Generation](https://nextjs.org/docs/basic-features/pages#static-generation-recommended) is not a good idea if you cannot pre-render a page ahead of a user's request. Maybe your page shows frequently updated data, and the page content changes on every request. ？？

Your page displays frequently updated data
![image](https://user-images.githubusercontent.com/32978177/155247461-9abadb97-0cbb-4cee-b90f-27dab5cd4315.png)

**Reply by parlay96** [reply]
@[balazsorban44](https://github.com/balazsorban44)   Thank you very much for your help

**Comment by parlay96** [comment]
Getstaticprops is static. We also need the client to update the data
![image](https://user-images.githubusercontent.com/32978177/155109460-63046d23-7ebe-4f62-9e2c-c2f9e385bce1.png)

**Comment by parlay96** [comment]
![image](https://user-images.githubusercontent.com/32978177/155255774-586dd8e8-fc93-4762-be53-f8921ff6195e.png)
getStaticProps  Unable to get query, req
But I want to get the address truncation parameter on the URL

![image](https://user-images.githubusercontent.com/32978177/155255960-bc0dc4ad-aca8-446c-b4d6-7863752204f3.png)
